clients/src/main/resources/message/ListOffsetRequest.json

```diff
@@ -0,0 +1,48 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+{
+  "apiKey": 2,
+  "type": "request",
+  "name": "ListOffsetRequest",
+  // Version 1 removes MaxNumOffsets.  From this version forward, only a single
+  // offset can be returned.
+  // Version 2 adds the isolation level, which is used for transactional reads.
+  // Version 3 is the same as version 2.
+  // Version 4 adds the current leader epoch, which is used for fencing.
+  "validVersions": "0-4",
+  "fields": [
+    { "name": "ReplicaId", "type": "int32", "versions": "0+",
+      "about": "The broker ID of the requestor, or -1 if this request is being made by a normal consumer." },
+    { "name": "IsolationLevel", "type": "int8", "versions": "2+",
```

Comment:
Yeah, enums would be a good addition.  It's a bit trickier than it seems, though.  For one thing, we already have a lot of public/stable enum types, and we can't break them.  enums can't inherit from other classes, either, so that escape hatch is closed.

Perhaps a simple way to make it work would be generating simple constants
```
public static int NONE = 0;
public static int OFFSET_OUT_OF_RANGE = 1;
... etc. etc. ...
```

Then the existing public/stable enums could be retrofitted to grab the values from these declarations.  We could also generate classes in cases where there was no existing public thing.